Good catch, you're right. The code floors at `max(d/2, minWait)`, so half the interval is the real floor and `minWait` only takes over when it's bigger than that. As written the comment would have someone expecting the at-cap wait to reach `minWait`, which it can't.

Reworded to "floored at half the interval, or at minWait when minWait exceeds half", and said explicitly that the `minWait` half only matters when `minWait` is more than half of `maxWait`.

Thanks for checking the bounds in both branches and the rand/v2 concurrency point. Pushed in 4ab50a5.

One note for whoever picks this up: the `integration-test (4.0, spark4)` failure is not from this change. The job dies in setup while pulling images, before any test runs:

```
rest Error unauthorized: authentication required
Error response from daemon: unauthorized: authentication required
make: *** [Makefile:46: integration-setup-spark4] Error 1
```

The commit it ran against only edits a doc comment, so it can't have broken an image pull. Looks like registry auth or rate limiting. A re-run should clear it.

--- a/catalog/hive/lock.go
+++ b/catalog/hive/lock.go
@@ -21,6 +21,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"math/rand/v2"
 	"slices"
 	"strings"
 	"time"
@@ -100,7 +101,11 @@ func acquireLocks(ctx context.Context, client HiveClient, identifiers []tableLoc
 	// If not acquired immediately, wait and retry
 	for attempt := 0; attempt < opts.LockRetries; attempt++ {
 		// Wait before checking again
-		waitTime := calculateBackoff(attempt, opts.LockMinWaitTime, opts.LockMaxWaitTime)
+		waitTime := applyJitter(
+			calculateBackoff(attempt, opts.LockMinWaitTime, opts.LockMaxWaitTime),
+			opts.LockMinWaitTime,
+			opts.LockMaxWaitTime,
+		)
 
 		select {
 		case <-ctx.Done():
@@ -158,6 +163,75 @@ func calculateBackoff(attempt int, minWait, maxWait time.Duration) time.Duration
 	}
 
 	return minWait << attempt
+}
+
+// applyJitter spreads a backoff interval by adding a random amount on top of it,
+// bounded so the result never exceeds maxWait.
+//
+// calculateBackoff is a pure function of the attempt number and the configured
+// bounds, so every client waiting on the same table lock computes an identical
+// sequence of delays. Contention is the precondition for entering the retry loop
+// at all, which means those clients are waiting simultaneously by construction:
+// they re-check the lock in lockstep, and each round of CheckLock calls arrives
+// at the metastore as a burst. Spreading the wait decorrelates them.
+//
+// The jitter is added rather than subtracted so that the result is never shorter
+// than the interval calculateBackoff produced. That keeps the guarantee implied
+// by the lock-check-min-wait-time property: a caller who configures a minimum
+// wait never polls sooner than it.
+//
+// Once the backoff saturates at maxWait there is no headroom left to add into.
+// calculateBackoff reaches that point deliberately, so leaving it unjittered
+// would put contending clients back in lockstep for every retry after the
+// sequence tops out. The wait is therefore spread downwards instead. Drawing
+// below maxWait breaks no contract, because it is an upper bound on the polling
+// interval rather than a target.
+//
+// How far down it may draw is the subtle part. Flooring at half the interval is
+// wrong: the last attempt that did not saturate waited for its own full interval,
+// which is somewhere in [maxWait/2, maxWait), so a floor of maxWait/2 lets the
+// first saturated attempt wait less than the attempt before it did. That inverts
+// the one property callers rely on, which is that the wait between lock checks
+// only ever grows. The floor is therefore the last interval the doubling sequence
+// produced before it hit the cap, recovered by replaying the sequence. It is a
+// bound the schedule has already cleared, so honouring it keeps the guaranteed
+// minimum monotonic across the saturation boundary, and it can never fall below
+// minWait because minWait is where the sequence starts.
+func applyJitter(d, minWait, maxWait time.Duration) time.Duration {
+	if d <= 0 {
+		return d
+	}
+
+	// A caller that hands in an interval already past the cap is outside the
+	// contract; leave it exactly as given rather than silently reshaping it.
+	headroom := maxWait - d
+	if headroom < 0 {
+		return d
+	}
+
+	// Add up to another full interval, without exceeding the configured maximum.
+	extra := d
+	if headroom < extra {
+		extra = headroom
+	}
+	if extra > 0 {
+		return d + time.Duration(rand.Int64N(int64(extra)+1))
+	}
+
+	// Replay the doubling sequence and keep the largest interval that still fitted
+	// under the cap. The guard on scheduled keeps a non-positive or overflowing
+	// minWait from spinning here; in that case the half-interval floor stands.
+	floor := d / 2
+	for scheduled := minWait; scheduled > 0 && scheduled < maxWait; scheduled <<= 1 {
+		if scheduled > floor {
+			floor = scheduled
+		}
+	}
+	if floor >= d {
+		return d
+	}
+
+	return d - time.Duration(rand.Int64N(int64(d-floor)+1))
 }
 
 func (l *HiveLock) Release(ctx context.Context) error {

--- a/catalog/hive/lock_test.go
+++ b/catalog/hive/lock_test.go
@@ -404,3 +404,133 @@ func TestLockConfigurationDefaults(t *testing.T) {
 	assert.Equal(t, DefaultLockCheckMaxWaitTime, opts.LockMaxWaitTime)
 	assert.Equal(t, DefaultLockCheckRetries, opts.LockRetries)
 }
+
+func TestApplyJitterBelowCapNeverShorterThanInput(t *testing.T) {
+	minWait := time.Millisecond
+	maxWait := time.Minute
+
+	for _, d := range []time.Duration{
+		time.Millisecond,
+		100 * time.Millisecond,
+		time.Second,
+	} {
+		for i := 0; i < 500; i++ {
+			got := applyJitter(d, minWait, maxWait)
+			assert.GreaterOrEqual(t, got, d,
+				"jitter must never wait less than the configured interval")
+			assert.LessOrEqual(t, got, min(2*d, maxWait),
+				"jitter must not exceed one extra interval or the configured maximum")
+		}
+	}
+}
+
+// Once calculateBackoff tops out there is no headroom to add into, so the wait
+// is spread downwards. It must still vary, or every retry after the sequence
+// saturates puts contending clients back in lockstep.
+func TestApplyJitterAtCapStaysWithinBoundsAndVaries(t *testing.T) {
+	minWait := 100 * time.Millisecond
+	maxWait := time.Second
+
+	seen := make(map[time.Duration]struct{})
+	for i := 0; i < 500; i++ {
+		got := applyJitter(maxWait, minWait, maxWait)
+		assert.GreaterOrEqual(t, got, maxWait/2,
+			"the spread at the cap must not exceed one half interval")
+		assert.LessOrEqual(t, got, maxWait,
+			"the wait must never exceed the configured maximum")
+		seen[got] = struct{}{}
+	}
+
+	assert.Greater(t, len(seen), 1,
+		"waits at the cap must vary so contending clients do not poll in lockstep")
+}
+
+// The spread at the cap must never poll sooner than a configured minimum wait,
+// which is reachable whenever minWait is more than half of maxWait.
+func TestApplyJitterAtCapNeverPollsSoonerThanMinWait(t *testing.T) {
+	minWait := 40 * time.Second
+	maxWait := time.Minute
+
+	for i := 0; i < 500; i++ {
+		assert.GreaterOrEqual(t, applyJitter(maxWait, minWait, maxWait), minWait,
+			"a caller who configures a minimum wait must never poll sooner than it")
+	}
+}
+
+// The guaranteed minimum must not fall as the sequence saturates. The attempt
+// before the cap waits for its own full interval, so flooring the spread at half
+// the cap would let the first saturated attempt poll sooner than the one before
+// it, which inverts the property that the wait between checks only ever grows.
+//
+// Reaching the cap takes a lowered lock-check-max-wait-time or a raised retry
+// count; the defaults (100ms, 1 minute, 4 retries) top out at 800ms and never
+// saturate. The values below are the smallest that put the boundary in range.
+func TestApplyJitterMinimumIsMonotonicAcrossTheCap(t *testing.T) {
+	minWait := 100 * time.Millisecond
+	maxWait := time.Second
+
+	// The sequence runs 100ms, 200ms, 400ms, 800ms, then saturates at 1s.
+	// Attempt 3 draws from [800ms, 1s]; attempt 4 is the first capped one.
+	//
+	// The bound compared against is exact, not sampled. Below the cap the jitter is
+	// added on top, so the interval itself is the floor and no estimate is needed.
+	// Sampling both sides instead would compare two noisy minima that sit a hair
+	// above the same true floor, and which of them lands lower is chance.
+	var floor time.Duration
+	for attempt := 0; attempt < 8; attempt++ {
+		d := calculateBackoff(attempt, minWait, maxWait)
+
+		for i := 0; i < 2000; i++ {
+			assert.GreaterOrEqual(t, applyJitter(d, minWait, maxWait), floor,
+				"attempt %d may not be allowed to wait less than an earlier attempt", attempt)
+		}
+
+		if d < maxWait {
+			floor = d
+		}
+	}
+}
+
+// The specific boundary above, pinned so a future change to the floor cannot
+// quietly reintroduce the dip without failing on the exact numbers.
+func TestApplyJitterAtCapFloorsAtTheLastUncappedInterval(t *testing.T) {
+	minWait := 100 * time.Millisecond
+	maxWait := time.Second
+
+	assert.Equal(t, 800*time.Millisecond, calculateBackoff(3, minWait, maxWait),
+		"the last interval before the cap")
+	assert.Equal(t, maxWait, calculateBackoff(4, minWait, maxWait),
+		"the first interval at the cap")
+
+	for i := 0; i < 2000; i++ {
+		assert.GreaterOrEqual(t, applyJitter(maxWait, minWait, maxWait), 800*time.Millisecond,
+			"the spread at the cap must not reach below the last uncapped interval")
+	}
+}
+
+func TestApplyJitterRespectsMaxWait(t *testing.T) {
+	minWait := 100 * time.Millisecond
+	maxWait := 1 * time.Second
+
+	for i := 0; i < 500; i++ {
+		// d above the cap should not be inflated further.
+		assert.Equal(t, 2*time.Second, applyJitter(2*time.Second, minWait, maxWait))
+	}
+}
+
+func TestApplyJitterVaries(t *testing.T) {
+	seen := make(map[time.Duration]struct{})
+	for i := 0; i < 500; i++ {
+		seen[applyJitter(time.Second, time.Millisecond, time.Minute)] = struct{}{}
+	}
+
+	// A deterministic implementation returns one value. The range here is a
+	// second wide, so observing a single value across 500 draws is not chance.
+	assert.Greater(t, len(seen), 1,
+		"waits must vary so that clients contending for the same lock do not poll in lockstep")
+}
+
+func TestApplyJitterNonPositive(t *testing.T) {
+	assert.Equal(t, time.Duration(0), applyJitter(0, time.Millisecond, time.Minute))
+	assert.Equal(t, -time.Second, applyJitter(-time.Second, time.Millisecond, time.Minute))
+}

